File: src/dataset/cpp/queen-attack_exercism-stungeye/claude/ClaudeSonnet4_queen_attack_v1.cpp
#include "queen_attack.h"

#include <stdexcept>

namespace queen_attack {
	chess_board::chess_board(position white, position black) : m_white{white}, m_black{black} {
		if (white == black) throw std::domain_error("Black and white positions cannot be the same.");
	}

	position chess_board::white() const {
		return m_white;
	}

	position chess_board::black() const {
		return m_black;
	}

	bool chess_board::can_attack() const {
		const int row_diff = m_white.first - m_black.first;
		const int col_diff = m_white.second - m_black.second;
		
		return row_diff == 0 
			   || col_diff == 0
			   || (row_diff * row_diff) == (col_diff * col_diff);
	}

	chess_board::operator std::string() const {
		std::string board(144, '_');
		
		// Set up row structure
		for (int i = 1; i < 144; i += 2) {
			if ((i + 1) % 18 == 0) {
				board[i] = '\n';
			} else {
				board[i] = ' ';
			}
		}
		
		board[m_white.first * 18 + m_white.second * 2] = 'W';
		board[m_black.first * 18 + m_black.second * 2] = 'B';

		return board;
	}
}  // namespace queen_attack

File: src/dataset/typescript/queen-attack_exercism-wackerow/claude/ClaudeSonnet4_queen-attack_v1.ts
type coords = [number, number];

interface Board {
  white: coords,
  black: coords
}

export default class Queens {
  private _white: coords;
  private _black: coords;

  constructor(private _board: Board) {
    this._white = _board.white;
    this._black = _board.black;
    
    if (this._white[0] === this._black[0] && this._white[1] === this._black[1]) {
      throw 'Queens cannot share the same space';
    }
  }

  get white(): coords {
    return [...this._white];
  }

  get black(): coords {
    return [...this._black];
  }

  public toString(): string {
    const whiteRow = this._white[0];
    const whiteCol = this._white[1];
    const blackRow = this._black[0];
    const blackCol = this._black[1];
    
    let result = '';
    for (let i = 0; i < 8; i++) {
      for (let j = 0; j < 8; j++) {
        if (i === whiteRow && j === whiteCol) {
          result += 'W';
        } else if (i === blackRow && j === blackCol) {
          result += 'B';
        } else {
          result += '_';
        }
        if (j < 7) result += ' ';
      }
      result += '\n';
    }
    return result;
  }

  public canAttack(): boolean {
    const [whiteRow, whiteCol] = this._white;
    const [blackRow, blackCol] = this._black;
    
    // Check same row or column
    if (whiteRow === blackRow || whiteCol === blackCol) {
      return true;
    }
    
    // Check diagonals
    const rowDiff = Math.abs(whiteRow - blackRow);
    const colDiff = Math.abs(whiteCol - blackCol);
    
    return rowDiff === colDiff;
  }
}